modules/platform/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/BridgeCommentManager.java

```diff
@@ -160,28 +168,34 @@ public Comment getComment(CoreSession session, String commentId)
     public Comment updateComment(CoreSession session, String commentId, Comment comment)
             throws CommentNotFoundException, CommentSecurityException {
         DocumentRef commentRef = new IdRef(commentId);
-        if (!session.exists(commentRef)) {
-            throw new CommentNotFoundException("The comment " + commentId + " does not exist");
-        }
-        if (session.getDocument(commentRef).getPropertyValue(COMMENT_PARENT_ID_PROPERTY) != null) {
-            return second.updateComment(session, commentId, comment);
-        } else {
-            return first.updateComment(session, commentId, comment);
-        }
+        return CoreInstance.doPrivileged(session, s -> {
+            // retrieve comment to check which service handles it
+            if (!s.exists(commentRef)) {
+                throw new CommentNotFoundException("The comment " + commentId + " does not exist");
+            }
+            if (s.getDocument(commentRef).getPropertyValue(COMMENT_PARENT_ID_PROPERTY) != null) {
+                return second.updateComment(session, commentId, comment);
+            } else {
+                return first.updateComment(session, commentId, comment);
+            }
+        });
```

Comment:
I am wondering in the case of migrating the comment from property to tree as the same implementation has the `COMMENT_PARENT_ID_PROPERTY`. perhaps we replace it by:

```
    @Override
    public Comment updateComment(CoreSession session, String commentId, Comment comment)
            throws CommentNotFoundException, CommentSecurityException {
        DocumentRef commentRef = new IdRef(commentId);
        return execute(session, commentRef, cm -> cm.updateComment(session, commentId, comment));
    }
```
wdyt and I did the change on my local machine it compile... :) but i didn't run the test

Reply:
I don't want to change the implementation there, If this has worked until there I would rather keep it. I just added a doPriviledge because only administrator case was tested before.